src/util/request-manager.js

```diff
@@ -428,6 +433,13 @@ export default class RequestManager {
           }
         }
 
+        const server = res.caseless.get('server');
+
+        if (res.statusCode === 401 && server === "GitHub.com") {
```

Comment:
Not sure if a constant is worth using here. Request manager is probably too generic in some ways. Would be nice to have a Github only request manager so stuff like this could be avoided. For now, I don't think this is terrible.